drake/solvers/Constraint.m

```diff
@@ -54,9 +56,9 @@
 
       obj.name = repmat({''},obj.num_cnstr,1);
 
-      obj.iCfun = reshape(bsxfun(@times,(1:obj.num_cnstr)',ones(1,obj.xdim)),[],1);
-      obj.jCvar = reshape(bsxfun(@times,1:obj.xdim,ones(obj.num_cnstr,1)),[],1);
-      obj.nnz = obj.num_cnstr*obj.xdim;
+      obj.iCfun = nan;
+      obj.jCvar = nan;
+      obj.nnz = nan;
```

Comment:
I don't particularly like having this special meaning of nan for this case; makes the code harder to read, and I feel like the constructor should do all of the work to construct an object, not leave some variables in a halfway constructed state, where you maybe have to call this additional method later to complete the construction process.

Maybe a better solution would be to add the sparsity pattern (as, say, a structure with fields iCfun, jCvar, and nnz) as an optional argument to the constructor of Constraint. If it's not passed in, use the fully dense matrices as the default as before, otherwise use the ones from the sparsity pattern struct.